src/main/scala/strawman/collection/immutable/SortedSet.scala

```diff
@@ -0,0 +1,21 @@
+package strawman.collection.immutable
+
+import strawman.collection.{SetMonoTransforms, Sorted, SortedLike}
+
+/** Base trait for sorted sets */
+trait SortedSet[A]
+  extends Set[A]
+    with Sorted[A]
+    with SortedSetLike[A, SortedSet] // Inherited SortedSet operations return a `SortedSet`
+
+trait SortedSetLike[A, +C[X] <: SortedSet[X]]
+  extends SortedLike[A, C]
+    with SetLike[A, Set] // Inherited Set operations return a `Set`
+    with SetOps[A, C[A]] // Override the return type of Set ops to return C[A]
+    with SetMonoTransforms[A, C[A]] {
+
+  // We have to write again the definition of methods inherited from SortedPolyTransforms
+  // so that they take precedence over the one inherited from IterablePolyTransforms
+  def map[B](f: A => B)(implicit ordering: scala.Ordering[B]): C[B]
+
+}
```

Comment:
One problem is that `Sorted` and `Set` are independent so there is no disambiguation mechanism between their overloaded methods. That’s why I had to re-define the methods of `Sorted` (to make them take precedence over those of `Set`).

I also tried an approach based on intermediate `LowPriority` linear traits, without success:

~~~ scala
trait SortedSetLikeLowPriority1[A, +C[X] <: SortedSet[X]]
  extends SetLike[A, Set]
    with SetOps[A, C[A]]
    with SetMonoTransforms[A, C[A]]

trait SortedSetLikeLowPriority2[A, +C[X] <: SortedSet[X]]
  extends SortedSetLikeLowPriority1[A, C]
    with SortedLike[A, C]

trait SortedSetLike[A, +C[X] <: SortedSet[X]]
  extends SortedSetLikeLowPriority2[A, C]
~~~